### DISCUSSION TITLE
Shape Resize doesn't keep new size

### DISCUSSION THREAD

**Discussion by jlhawkins-stuff** [eclipse-glsp/glsp#438] [Q&A] [answered]
I am running the theia integration workflow example in eclipse che theia and resizing shapes don't keep the new size.
The minimal example works as expected.
Is that the expected behavior for the workflow example?  And what code would allow the resizing to persist?

**Comment by tortmayr** [comment, marked as answer]
Hi @jlhawkins-stuff, this behavior is caused by a known bug (#219). Unfortunately, there is no quick fix available and changes in the base framework are required. This is issue on the top of our backlog and we hopefully can provide a fix soon.

**Reply by jlhawkins-stuff** [reply]
Thank you for letting me know about that.  

While I was debugging the server (GLSP Java), I noticed the bounds operations sequences.  

When the shape was resized, the server initially received an apply bounds action.  Within that message contained the dimensions of the shape, after the hint was released.  

The next message/action I noticed the server receiving was a change bounds operation.  That message did not have any dimension values in it.  When that method completed the shape on the diagram would return to its original size.  

When saving the diagram I noticed the diagram model would retain the shape size in the layout object properties but the size properties would remain at the initial values set in the operations handler.

Should the framework glsp-client node_module package be sending the changed dimensions in the change bounds message also?

**Reply by CamilleLetavernier** [reply]
> Should the framework glsp-client node_module package be sending the changed dimensions in the change bounds message also?

It mostly depends on what the layout expects. In #392 / eclipse-glsp/glsp-client/pull/136, we added a new version of the VBox layout, that can take both the manual size (Set by a ChangeBoundsOperation) and the computed size (Derived from the figure & children, e.g. text size or children-size), and will apply the highest value. This allows automatic-resizing of nodes, as well as manual-resizing at the same time; whereas the previous VBox layout only allowed one or the other.

Applying this layout to other nodes, and changing the type hints to make these nodes resizable, should fix the bug #219 (In theory :) )

> I am running the theia integration workflow example in eclipse che theia and resizing shapes don't keep the new size.

Which nodes are you referring to, exactly? IIRC, there are currently 5 resizable nodes: Fork/Join/Merge/Decision, and Categories. I believe they should all work properly (Fork/Join/Merge/Decision don't use micro-layouting, and Category was implemented specifically with the new layout). Tasks don't work well with manual resizing at the moment, but the resize feature should be disabled on these nodes anyway.

**Reply by jlhawkins-stuff** [reply]
@CamilleLetavernier Attached is the workflow-example:
![workflow-example-resize](https://user-images.githubusercontent.com/19364352/139503098-c7c19049-8470-4db7-8435-4a3629ec4fca.gif)

And then here's the minimal-example:
![minimal-example-resize](https://user-images.githubusercontent.com/19364352/139503251-f8bec206-0a9b-4a8d-ac27-c60d6799ad46.gif)

Here's a basic example of a project I'm working on using the latest packages.
![basic-example-rebuild-resize](https://user-images.githubusercontent.com/19364352/139503411-d177a3bb-741d-477a-abcf-9d8619c48adf.gif)

**Comment by tortmayr** [comment]
@jlhawkins-stuff  I can reproduce your issue and after investigating this it seems like we introduced a bug with a recent PR. I opened #441 to track this. Thank you for your efforts in finding that bug!

Update: A [fix](https://github.com/eclipse-glsp/glsp-server/pull/139) for this is currently under review.

**Comment by tortmayr** [comment]
@jlhawkins-stuff  Resize should now work again expected.
